src/Simulator/IncrementalRedstoneSimulator/ObserverHandler.h

```diff
@@ -0,0 +1,106 @@
+
+#pragma once
+
+#include "RedstoneHandler.h"
+#include "../../Blocks/BlockObserver.h"
+
+
+
+
+
+class cObserverHandler : public cRedstoneHandler
+{
+public:
+
+	inline static bool IsOn(NIBBLETYPE a_Meta)
+	{
+		return (a_Meta & 0x8) == 0x8;
+	}
+
+	static bool ShouldPowerOn(cWorld & a_World, const Vector3i & a_Position, NIBBLETYPE a_Meta, cIncrementalRedstoneSimulatorChunkData * a_Data)
```

Comment:
`Vector3i` should be passed by-value (compiler places its components into registers).